frontend/javascripts/admin/task/task_search_form.js

```diff
@@ -99,16 +104,13 @@ class TaskSearchForm extends React.Component<Props, State> {
     this.setState({ users, projects, taskTypes, isFetchingData: false });
   }
 
-  handleFormSubmit = (
-    isRandom: boolean,
-    onFinishCallback: QueryObject => Promise<void>,
-    event: ?SyntheticInputEvent<*>,
-  ) => {
-    if (event) {
-      event.preventDefault();
+  handleFormSubmit = (isRandom: boolean, onFinishCallback: QueryObject => Promise<void>) => {
+    const form = this.formRef.current;
+    if (!form) {
+      return;
     }
 
-    this.props.form.validateFields((err, formValues: TaskFormFieldValues) => {
+    form.validateFields().then(formValues => {
```

Comment:
Looks too me like this can be done with `onFinish` not `onSubmit`. That way this method would be smaller and easier to understand.
Using `onFinish` ensures that all fields are valid and therefor a `form.validateFields()` check is not needed